src/sentry/workflow_engine/endpoints/validators.py

```diff
@@ -0,0 +1,148 @@
+from django.db import router, transaction
+from rest_framework import serializers
+from rest_framework.fields import Field
+
+from sentry import audit_log
+from sentry.api.serializers.rest_framework import CamelSnakeSerializer
+from sentry.issues import grouptype
+from sentry.issues.grouptype import GroupType
+from sentry.utils.audit import create_audit_entry
+from sentry.workflow_engine.models import DataConditionGroup, Detector
+from sentry.workflow_engine.models.data_condition import Condition, DataCondition
+from sentry.workflow_engine.types import DetectorPriorityLevel
+
+
+class BaseDataConditionValidator(CamelSnakeSerializer):
+
+    condition = serializers.CharField(
+        required=True,
+        max_length=200,
+        help_text="Condition used to compare data value to the stored comparison value",
+    )
+
+    @property
+    def comparison(self) -> Field:
+        raise NotImplementedError
+
+    @property
+    def condition_result(self) -> Field:
+        raise NotImplementedError
+
+    @property
+    def type(self) -> str:
+        # TODO: This should probably at least be an enum
+        raise NotImplementedError
+
+    def validate(self, attrs):
+        attrs = super().validate(attrs)
+        attrs["type"] = self.type
+        return attrs
+
+
+class NumericComparisonConditionValidator(BaseDataConditionValidator):
+
+    comparison = serializers.FloatField(
+        required=True,
+        help_text="Comparison value to be compared against value from data.",
+    )
+    result = serializers.ChoiceField(
+        choices=[
+            (DetectorPriorityLevel.HIGH, "High"),
+            (DetectorPriorityLevel.MEDIUM, "Medium"),
+            (DetectorPriorityLevel.LOW, "Low"),
+        ]
+    )
+
+    @property
+    def supported_conditions(self) -> frozenset[Condition]:
+        raise NotImplementedError
+
+    @property
+    def supported_results(self) -> frozenset[DetectorPriorityLevel]:
+        raise NotImplementedError
+
+    def validate_condition(self, value: str) -> Condition:
+        try:
+            condition = Condition(value)
+        except ValueError:
+            condition = None
+
+        if condition not in self.supported_conditions:
+            raise serializers.ValidationError(f"Unsupported condition {value}")
+        return condition
+
+    def validate_result(self, value: str) -> DetectorPriorityLevel:
+        try:
+            result = DetectorPriorityLevel(int(value))
+        except ValueError:
+            result = None
+        if result not in self.supported_results:
+            raise serializers.ValidationError("Unsupported condition result")
+        return result
+
+
+class BaseDataSourceValidator(CamelSnakeSerializer):
+    pass
+
+
+class BaseGroupTypeDetectorValidator(CamelSnakeSerializer):
```

Comment:
Thoughts on naming this something like this instead?
```suggestion
class BaseConditionValidator(CamelSnakeSerializer):
```

I feel like the more important aspect of the work happening here is related to the DataConditions rather than the GroupType, but it does require the group_type to be correct as well.. 🤔

Reply:
It's specifically a validator for accepting an entire Detector that is customized by group type. The conditions are a small part